tests/functional/pod_and_daemons/test_cephtoolbox_pod_nodeaffinity.py

```diff
@@ -0,0 +1,130 @@
+import logging
+import pytest
+
+from ocs_ci.framework import config
+from ocs_ci.framework.pytest_customization.marks import bugzilla, brown_squad
+from ocs_ci.framework.testlib import tier1, tier4b, polarion_id
+from ocs_ci.helpers.sanity_helpers import Sanity
+from ocs_ci.ocs import ocp, constants
+from ocs_ci.ocs.node import (
+    unschedule_nodes,
+    drain_nodes,
+    schedule_nodes,
+    taint_nodes,
+    untaint_nodes,
+    get_worker_nodes,
+    get_worker_node_where_ceph_toolbox_not_running,
+    apply_node_affinity_for_ceph_toolbox,
+)
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs.resources.pod import (
+    wait_for_pods_to_be_running,
+)
+
+log = logging.getLogger(__name__)
+
+
+@brown_squad
+@bugzilla("2249640")
+class TestCephtoolboxPod:
+    @pytest.fixture(autouse=True)
+    def init_sanity(self):
+        """
+        Initialize Sanity instance
+        """
+        self.sanity_helpers = Sanity()
+
+    @pytest.fixture(scope="session", autouse=True)
+    def teardown(self, request):
+        def finalizer():
+            untaint_nodes()
+            resource_name = constants.DEFAULT_CLUSTERNAME
+            if config.DEPLOYMENT["external_mode"]:
+                resource_name = constants.DEFAULT_CLUSTERNAME_EXTERNAL_MODE
+            storagecluster_obj = ocp.OCP(
+                resource_name=resource_name,
+                namespace=config.ENV_DATA["cluster_namespace"],
+                kind=constants.STORAGECLUSTER,
+            )
+            params = '[{"op": "remove", "path": "/spec/placement/toolbox"},]'
+            storagecluster_obj.patch(params=params, format_type="json")
+            log.info("Patched storage cluster  back to the default")
+            assert (
+                wait_for_pods_to_be_running()
+            ), "some of the pods didn't came up running"
+
+        request.addfinalizer(finalizer)
+
+    @tier1
+    @polarion_id("OCS-6086")
+    def test_node_affinity_to_ceph_toolbox_pod(self):
+        """
+        This test verifies whether ceph toolbox failovered or not after applying node affinity
+        """
+        other_nodes = get_worker_node_where_ceph_toolbox_not_running()
+        log.info(
+            "Apply node affinity with a node name other than currently running node."
+        )
+        assert apply_node_affinity_for_ceph_toolbox(other_nodes[0])
+
+    @tier4b
+    @polarion_id("OCS-6087")
+    def test_reboot_node_affinity_node(self):
+        """
+        This test verifies ceph toolbox runs only on the node given in node-affinity.
+        Reboot the node after applying node-affinity.
+        Expectation is the pod should come up only on that node mentioned in affinity.
+        """
+        other_nodes = get_worker_node_where_ceph_toolbox_not_running()
+        node_name = other_nodes[0]
```

Comment:
instead of hard coding with a same node every time, randomly select a ceph pod running worker node so that it picks a random node everytime